src/components/dialog/DialogBase.tsx

```diff
@@ -81,7 +81,12 @@ export const DialogBase = ({
             onEscKeyDown={onEscKeyDown}
           >
             <FocusRing>
-              <Element ref={ref} data-teamleader-ui="dialog" className={dialogClassNames} {...(form && { onSubmit })}>
+              <Element
+                ref={ref as RefObject<HTMLFormElement> & RefObject<HTMLDivElement>}
```

Comment:
This needs to be an `|` operator instead of an `&`, no?

Reply:
I've changed the way this element is rendered into a form or div a bit, since TS really wanted the ref's type to be `RefObject<HTMLFormElement> & RefObject<HTMLDivElement>`